packages/firestore/src/remote/remote_store.ts

```diff
@@ -255,10 +263,22 @@ export class RemoteStore implements TargetMetadataProvider {
   }
 
   private startWatchStream(): void {
-    assert(
-      this.shouldStartWatchStream(),
-      'startWriteStream() called when shouldStartWatchStream() is false.'
-    );
+    this.startWatchStreamAfterError(/* afterError= */ false);
```

Comment:
Having a boolean argument that negates the method name is confusing to me.

An improvement would be to have a startWatchStreamInternal() method. And then startWatchStream() and startWatchStreamAfterError() can perform the appropriate assertion and then call startWatchStreamInternal().

But at this point I'd question what value we are getting from having 2 different startWatchStream() methods that are identical except in the assertion they perform... and whether that outweighs the conceptual complexity of having separate methods to do the same thing. I'd probably opt to simplify or remove the assertion instead.